src/lib/utils/error-logger.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const logger = {};
+
+const getLogData = (err) => {
+    let logData;
+    if (err.stack) {
+        logData = err.stack;
+    } else if (err.message) {
+        logData = err.message;
+    } else if (err.toString instanceof Function) {
+        logData = err.toString();
+    } else {
+        logData = err;
+    }
+
+    return logData;
+};
+
+module.exports.register = (server, options, next) => {
+
+    //logs error
+    logger.logError = (tags,err) => {
+        server.log(['error'].concat(tags), getLogData(err));
+    };
+
+    //logs and throws error
+    logger.logAndThrowError = (tags,err) => {
```

Comment:
Thanks for the feedback. The server does catch and log unhandled errors. These logging functions are for handled errors. 

`logAndThrow` is useful if we catch an anticipated error, and want to log it, then pass the error up the stack to be handled further (or not). 

In your example, you advocate that _utility B_ be written such that it *relies* on its consumer (_utility A_) to perform logging that is more detailed than the basic 'unhandled error' logging provided by the server. I respectfully disagree: if an error is caught within _utility B_, then _utility B_ should log information that is useful in debugging that error. In the event that _utility B_ is expected to throw an error on failure, then _utility B_ should re-throw the error so that its consumers can handle it as appropriate.

To give some specific context here: in reality, _utility B_ is an authentication utility that retrieves and compares a user's password. In the event that the passwords do not match, or if no password can be found, the utility throws a boom-wrapped unauthorized error. In order to throroughly log the event, the authentication utility also logs the error, then re-throws it so that the consumer (the login route handler) can `reply` with it.